src/AxelarGateway.sol

```diff
@@ -161,6 +168,29 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         }
     }
 
+    function _mintTokenAndApproveContractCall(
```

Comment:
Can we merge `_approveContractCall` and `_mintTokenAndApproveContractCall` into one function to save some gas on deploying the gateway? Also, I don't think we need to events like `MintedAndContractCallApproved` and `ContractCallApproved`. One can already figure out that some token was minted in the same tx as `_mintToken` emits a `Transfer` event. Just `event ContractCallApproved(address contractAddress, ContractCallHashKey hashKey, bytes32 payloadHash);` should be sufficient.

Reply:
I think it's a good idea. I was originally trying to be explicit about what is happening, especially if we have many mint commands in one `_execute` transaction. But you are right, it is possible to determine previous mint event from the tx event list